webpack/assets/javascripts/react_app/components/common/DateTimePicker/DateTimePicker.js

```diff
@@ -101,13 +107,7 @@ class DateTimePicker extends React.Component {
           </OverlayTrigger>
           {!required && (
             <InputGroup.Addon className="clear-button">
-              <Icon
-                type="fa"
-                name="close"
-                onClick={() =>
-                  this.setState({ hiddenValue: true, value: new Date() })
-                }
-              />
+              <Icon type="fa" name="close" onClick={this.clearSelected} />
```

Comment:
We need to wrap call to anonymous function, like this we would pass around a pointer, that could change - I don't understand exacly why, but this is not safe by React guides ^_^
```suggestion
              <Icon type="fa" name="close" onClick={() => this.clearSelected} />
```

Reply:
I don't think this is necessary.. Since `<Icon>` is getting a new anonymous function as a prop, wouldn't it cause a rerender of `<Icon>` every time?  Since the function is anonymous, React is unable to see if the prop changed and so will just rerender no matter what.  Rather, if left as is, `this.clearSelected` won't have changed and so may avoid rerender of `<Icon>` every time.